tensorflow/tools/ci_build/ci_parameterized_build.sh

```diff
@@ -0,0 +1,204 @@
+str_strip () {
+  echo -e "$1" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
+}
+
+
+##########################################################
+# Default configuration
+CI_BUILD_DIR="tensorflow/tools/ci_build"
+
+# Command to call when Docker is available
+DOCKER_MAIN_CMD="${CI_BUILD_DIR}/ci_build.sh"
+# Command to call when Docker is unavailable
+NO_DOCKER_MAIN_CMD="${CI_BUILD_DIR}/builds/configured"
+
+# Additional option flags to apply when Docker is unavailable (e.g., on Mac)
+NO_DOCKER_OPT_FLAG="-j 1 --genrule_strategy=standalone "\
+"--linkopt=-headerpad_max_install_names"
+
+BAZEL_CMD="bazel test"
+PIP_CMD="${CI_BUILD_DIR}/builds/pip.sh"
+ANDROID_CMD="${CI_BUILD_DIR}/builds/android.sh"
+
+BAZEL_TARGET="//tensorflow/..."
+##########################################################
+
+# Convert all the required environment variables to lower case
+TF_BUILD_CONTAINER_TYPE=$(to_lower ${TF_BUILD_CONTAINER_TYPE})
+TF_BUILD_PYTHON_VERSION=$(to_lower ${TF_BUILD_PYTHON_VERSION})
+TF_BUILD_IS_OPT=$(to_lower ${TF_BUILD_IS_OPT})
+TF_BUILD_IS_PIP=$(to_lower ${TF_BUILD_IS_PIP})
+
+# Print parameter values
+echo "Required build parameters:"
+echo "  TF_BUILD_CONTAINER_TYPE=${TF_BUILD_CONTAINER_TYPE}"
+echo "  TF_BUILD_PYTHON_VERSION=${TF_BUILD_PYTHON_VERSION}"
+echo "  TF_BUILD_IS_OPT=${TF_BUILD_IS_OPT}"
+echo "  TF_BUILD_IS_PIP=${TF_BUILD_IS_PIP}"
+echo "Optional build parameters:"
+echo "  TF_BUILD_DRY_RUN=${TF_BUILD_DRY_RUN}"
+echo "  TF_BUILD_APPEND_CI_DOCKER_EXTRA_PARAMS="\
+"${TF_BUILD_APPEND_CI_DOCKER_EXTRA_PARAMS}"
+echo "  TF_BUILD_APPEND_ARGUMENTS=${TF_BUILD_APPEND_ARGUMENTS}"
+echo "  TF_BUILD_BAZEL_TARGET=${TF_BUILD_BAZEL_TARGET}"
+
+# Process container type
+CTYPE=${TF_BUILD_CONTAINER_TYPE}
+OPT_FLAG=""
+if [[ ${CTYPE} == "cpu" ]]; then
+  :
+elif [[ ${CTYPE} == "gpu" ]]; then
+  OPT_FLAG="--config=cuda"
+elif [[ ${CTYPE} == "android" ]]; then
+  :
+else
+  echo "Unrecognized value in TF_BUILD_CONTAINER_TYPE: "\
+"\"${TF_BUILD_CONTAINER_TYPE}\""
+  exit 1
+fi
+
+EXTRA_PARAMS=""
+
+# Determine if Docker is available
+MAIN_CMD=${DOCKER_MAIN_CMD}
+if [[ -z "$(which docker)" ]]; then
+  echo "It appears that Docker is not available on this system. "\
+"Will perform build without Docker."
+  echo "In addition, the additional option flags will be applied to the build:"
+  echo "  ${NO_DOCKER_OPT_FLAG}"
+  MAIN_CMD=${NO_DOCKER_MAIN_CMD}
+  OPT_FLAG="${OPT_FLAG} ${NO_DOCKER_OPT_FLAG}"
+
+elif [[ ${CTYPE} == "gpu" ]]; then  # Docker GPU: Map devices and libraries
+  CUDA_DEVICES=$(ls /dev/nvidia* | xargs -I{} echo '--device {}:{}' | xargs)
+  CUDA_LIBS=$(ls /usr/lib/x86_64-linux-gnu/libcuda* | xargs -I{} echo '-v {}:{}' | xargs)
+
+  EXTRA_PARAMS="${EXTRA_PARAMS} ${CUDA_DEVICES[@]} ${CUDA_LIBS[@]}"
+fi
+
+# Process Bazel "-c opt" flag
+if [[ ${TF_BUILD_IS_OPT} == "no_opt" ]]; then
+  # PIP builds are done only with the -c opt flag
+  if [[ ${TF_BUILD_IS_PIP} == "pip" ]]; then
+    echo "Skipping parameter combination: ${TF_BUILD_IS_OPT} & "\
```

Comment:
In fact, if you look at pip.sh (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/builds/pip.sh), the "-c opt" option is always used when running the target "//tensorflow/tools/pip_package:build_pip_package". So specifying NO_OPT here would have no effect anyway.